### ISSUE TITLE
Moving resource loses associated public shares

### ISSUE THREAD

**Issue by PVince81** [owncloud/ocis-reva#335]
This is using the OC storage, I haven't tested with EOS.

Scenario: tests/acceptance/features/apiSharePublicLink2/multilinkSharing.feature:180

Basically:

### Steps

1. Create folder "test"
1. Create public links for "test"
1. Rename "test" to "test_renamed"
1. Check the public link list for "test_renamed"

### Expected
Links are still there

### Actual
Links disappeared and are likely still linked to the other path ?!
Or maybe a cache issue ?

I thought shares were associated by resource id.

@refs

**Comment by PVince81** [comment]
I've checked the file id of a resource after renaming and it did not change.

However I find it strange that the fileid of a folder looks like "MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU0OjkzYTQxYTU3LTI0YzQtNDQ1Yy04ZDVkLTJiZDYxYmFiODY3MA==" but the value returned in "item_source" looks more like an id c72e3d65-0782-488f-8708-0ca70b84d00a

We should normalize this because clients expect to be able to use those values interchangeably. Raised here https://github.com/owncloud/ocis-reva/issues/344

**Comment by PVince81** [comment]
the problem with losing after rename also happens with user shares

**Comment by PVince81** [comment]
I've checked the values in redis and it appears that the mapping is done on path, not resource id:
```
1) "c72e3d65-0782-488f-8708-0ca70b84d00a"
127.0.0.1:6379> get c72e3d65-0782-488f-8708-0ca70b84d00a
"/var/tmp/reva/data/einstein/files/demos-2020-07-03
```

this is the resource I renamed, so its path changed.

we need to change the cached values to use resource ids instead of paths

**Comment by PVince81** [comment]
hmmm... but for caching purposes I guess it's fine. this is the file id, not share id.

when renaming we should discard the cached id to avoid path collisions, and then cache the new value.
and in case of cache miss, resolve the value to the correct path.

it seems that so far none of this is working, maybe not implemented yet ?

@butonic

**Comment by PVince81** [comment]
Indeed, I see that resolution of the path is simply failing instead of rescanning: https://github.com/cs3org/reva/blob/master/pkg/storage/fs/owncloud/owncloud.go#L545

**Comment by PVince81** [comment]
@individual-it has confirmed that this works on EOS, so this issue is specific to the OC storage. Looking at the source code also confirmed this, as Redis seems to be used by that storage.

**Comment by individual-it** [comment]
Works on EOS storage, I've enabled the test to be run on EOS storage https://github.com/owncloud/core/pull/37652
